### PR TITLE
Spawn bitcoind, lightningd, and lnd via Rust

### PR DESCRIPTION
**Pull request by justinmoon**
First step towards https://github.com/fedimint/fedimint/issues/812

Builds on https://github.com/fedimint/fedimint/pull/1870

Working with our shell scripts is miserable. Rust is much nicer. The idea here is to just start re-writing the shell-based integration tests piece-by-piece until it's all Rust.

Very open to ideas on how to structure this.

### CODE REVIEW

File: scripts/setup-tests.sh [hunk 1]
```diff
@@ -9,11 +9,13 @@ FM_FED_SIZE=${1:-4}
 source ./scripts/build.sh $FM_FED_SIZE
 
 # start daemons
-start_bitcoind
+rust_fixtures
```
Comment (dpc) [inline scripts/setup-tests.sh:12]: `start_fixtures` seems to fit better

**Review by dpc — APPROVED** [review]

File: scripts/setup-tests.sh (hunk 1 above)
Comment (justinmoon) [reply scripts/setup-tests.sh]: Fixed.

**Review by dpc — APPROVED** [review]

**Review by elsirion — DISMISSED** [review]
Really cool, but somehow the [nix container build is now complaining](https://github.com/fedimint/fedimint/actions/runs/4396206603/jobs/7698497952#step:7:2693) … is this related to #1873?

**Comment by justinmoon** [comment]
> Really cool, but somehow the [nix container build is now complaining](https://github.com/fedimint/fedimint/actions/runs/4396206603/jobs/7698497952#step:7:2693) … is this related to #1873?

I don't know why container build is complaining ...

**Comment by justinmoon** [comment]
By moving the binary from `integrationtests` to `fedimint-testing` I got the container build to succeed again. I think this makes more sense, because `integrationtests` contains tests and `fedimint-testing` contains helpers and this is basically a test helper ...

File: scripts/lib.sh [hunk 2]
```diff
@@ -146,25 +146,59 @@ function show_verbose_output()
...
+  await_bitcoin_rpc
   # create a default RPC wallet
   $FM_BTC_CLIENT createwallet ""
   # mine some blocks
-  mine_blocks 101 | show_verbose_output
+  mine_blocks 101
+  echo "started bitcoind"
 }
 
 function start_lightningd() {
```
Comment (elsirion) [inline scripts/lib.sh:200]: What's preventing us from getting rid of this function altogether now (same for lnd+bitcoind)?

**Review by elsirion — APPROVED** [review]

**Review by dpc — APPROVED** [review]

File: scripts/lib.sh (hunk 2 above)
Comment (justinmoon) [reply scripts/lib.sh:200]: Tmuxinator uses it. Next PR will get rid of all these functions.

**Pull request merged**